### DISCUSSION TITLE
Setup troubles

### DISCUSSION THREAD

**Discussion by SarreqTeryx** [meisnate12/Plex-Meta-Manager#1651] [Q&A]
I'm trying to setup PMM for the first time, following the Local Walkthrough, and I'm getting stuck on the config.yml section where it makes an intentional error with Movies-NOSUCHLIBRARY:
I'm not getting the expected error text, but this instead:
`Traceback (most recent call last):
  File "C:\Users\Sarreq Teryx\Plex-Meta-Manager\plex_meta_manager.py", line 173, in <module>
    from modules.builder import CollectionBuilder
  File "C:\Users\Sarreq Teryx\Plex-Meta-Manager\modules\builder.py", line 2, in <module>
    from arrapi import ArrException
  File "C:\Users\Sarreq Teryx\Plex-Meta-Manager\pmm-venv\Lib\site-packages\arrapi\__init__.py", line 1, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'`

any ideas?  I'm sure I followed the walkthrough exactly up to there.

**Comment by chazlarson** [comment]
What version of python is this?

**Reply by SarreqTeryx** [reply]
3.12.0

**Reply by chazlarson** [reply]
~~I can't reproduce this on a linux machine:~~

When I take that next step it indeed happens.  Whoops.

Get python 3.12:
```
> pyenv install 3.12
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
Installed Python-3.12.0 to /home/chaz/.pyenv/versions/3.12.0
```

Get the PMM code:
```
> git clone https://github.com/meisnate12/Plex-Meta-Manager
> cd Plex-Meta-Manager

Cloning into 'Plex-Meta-Manager'...
```
set up a virtual environment that uses Python 3.12
```
> echo 'layout pyenv 3.12.0' > .envrc
> direnv allow
direnv: loading ~/Plex-Meta-Manager/.envrc

> python --version
Python 3.12.0
```
Install requirements:
```
> python -m pip install -r requirements.txt
...
Successfully installed GitPython-3.1.36 PlexAPI-4.15.2 arrapi-1.4.3 certifi-2023.7.22 charset-normalizer-3.3.0 docopt-0.6.2 gitdb-4.0.10 idna-3.4 lxml-4.9.3 num2words-0.5.12 pathvalidate-3.2.0 pillow-10.0.1 psutil-5.9.5 python-dotenv-1.0.0 requests-2.31.0 retrying-1.3.4 ruamel.yaml-0.17.32 schedule-1.2.0 six-1.16.0 smmap-5.0.1 tmdbapis-1.2.0 urllib3-2.0.6
```
Run script
```
> python plex_meta_manager.py -r
Config Error: config not found at /home/chaz/Plex-Meta-Manager/config
```
Take the other steps
```
> mkdir config/assets
> cp config/config.yml.template config/config.yml
> nano config/config.yml
> python plex_meta_manager.py -r
Traceback (most recent call last):
  File "/home/chaz/Plex-Meta-Manager/plex_meta_manager.py", line 173, in <module>
    from modules.builder import CollectionBuilder
  File "/home/chaz/Plex-Meta-Manager/modules/builder.py", line 2, in <module>
    from arrapi import ArrException
  File "/home/chaz/Plex-Meta-Manager/.direnv/python-3.12.0/lib/python3.12/site-packages/arrapi/__init__.py", line 1, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

That's weird.

**Reply by chazlarson** [reply]
Python 3.11.6 works for me, if that's a possiblity.

**Reply by chazlarson** [reply]
Also, running `pip install setuptools` inside the 3.12.0 venv fixed it.

**Comment by chazlarson** [comment]
Seems like maybe something's wrong with the Python installation:
<https://stackoverflow.com/questions/7446187/no-module-named-pkg-resources>

Though admittedly that is pretty old, it matches the symptom.

Following that article's advice and doing `pip install setuptools` inside the 3.12.0 venv fixed this for me.
